### PR TITLE
Initial creation of azure deployment helper powershell

### PR DESCRIPTION
**Pull request by BlythMeister**
Initially using `Azure-DeployZipToWebApp` to deploy a zip folder to a webapp as the only supported functionality

### CODE REVIEW

File: src/Scripts/azureHelper.ps1 [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            exit $LASTEXITCODE
+        }
+    }
+    else
+    {
+        Write-Host "Already Logged In"
+    }
+}
+
+function Azure-DeployWebApp([string]$resourceGroup, [string]$name, [string]$zipPath)
```
Comment (richard-green) [inline src/Scripts/azureHelper.ps1:66]: These two functions feel like they're named backwards - this one deploys a zip to azure, and the other deploys a folder (which of course creates a zip and calls this function)

Comment (BlythMeister) [reply src/Scripts/azureHelper.ps1:66]: Your right, i'll flip the names.

File: src/Scripts/azureHelper.ps1 [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+function Azure-DeployZipToWebApp([string]$resourceGroup, [string]$name, [string]$zipPath)
+{
+    & az webapp deployment source config-zip --resource-group $resourceGroup --name $name --src $zipPath
+}
+
+function Azure-DeployWebApp([string]$username, [string]$password, [string]$tenant, [string]$resourceGroup, [string]$name, [string]$contentFolder)
+{
+    Azure-EnsureProfileActive $username $tenant
```
Comment (richard-green) [inline src/Scripts/azureHelper.ps1:73]: Redundant as the LoginServicePrincipal does this

File: src/Scripts/azureHelper.ps1 [hunk 3]
```diff
@@ -0,0 +1,88 @@
+﻿if($deployHelpLoaded -eq $null)
```
Comment (richard-green) [inline src/Scripts/azureHelper.ps1:1]: whitespace needed after if

File: src/Scripts/azureHelper.ps1 (hunk 2 above)
Comment (BlythMeister) [reply src/Scripts/azureHelper.ps1:73]: Correct, but it's possible to call this function direct, you don't need to use the helper function that does login then deploy,

File: src/Scripts/azureHelper.ps1 [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+Write-Host "Ensconce - AzureHelper Loading"
+if (Get-Command "az" -ErrorAction SilentlyContinue)
+{
+    $raw = (az version 2>&1) -join ""
+    $data = ConvertFrom-Json $raw
+    $cliVersion = $data.'azure-cli'
+    write-host "Azure CLI Version: $cliVersion"
+
```
Comment (richard-green) [inline src/Scripts/azureHelper.ps1:19]: excess whitespace

File: src/Scripts/azureHelper.ps1 [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    if(Test-Path "$contentFolder.zip")
+    {
+        Remove-Item "$contentFolder.zip" -Force
+    }
+    
+    CreateZip $contentFolder "$contentFolder.zip"
+    
+    Azure-DeployWebApp $resourceGroup $name "$content.zip"
+}
+
```
Comment (richard-green) [inline src/Scripts/azureHelper.ps1:86]: excess whitespace

File: src/Scripts/azureHelper.ps1 [hunk 6]
```diff
@@ -0,0 +1,88 @@
+﻿if($deployHelpLoaded -eq $null)
+{
+	$DeployToolsDir = Split-Path ((Get-Variable MyInvocation -Scope 0).Value.MyCommand.Path)
+    . $DeployToolsDir\deployHelp.ps1
+}
+
+if($compressionHelperLoaded -eq $null)
```
Comment (richard-green) [inline src/Scripts/azureHelper.ps1:7]: whitespace needed after if

File: src/Scripts/azureHelper.ps1 [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    throw "azure CLI not installed"
+}
+
+$rootProfilePath = "$Home\.azure-ensconce-profiles"
+
+function Azure-EnsureProfileActive([string]$username, [string]$tenant)
+{
+    $azureProfileId = "$username`_$tenant"
+    $azureProfilePath = [IO.Path]::Combine($rootProfilePath, $azureProfileId)
+    if($env:AZURE_CONFIG_DIR -ne $azureProfilePath)
```
Comment (richard-green) [inline src/Scripts/azureHelper.ps1:32]: whitespace needed after if

File: src/Scripts/azureHelper.ps1 [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    {
+        $env:AZURE_CONFIG_DIR = $azureProfilePath
+        Write-Host "Profile config set to $env:AZURE_CONFIG_DIR"
+    }
+}
+
+function Azure-LoginServicePrincipal([string]$username, [string]$password, [string]$tenant)
+{
+    Azure-EnsureProfileActive $username $tenant
+    if($env:AZURE_CONFIG_DIR -eq $null -or $env:AZURE_CONFIG_DIR -eq "")
```
Comment (richard-green) [inline src/Scripts/azureHelper.ps1:42]: whitespace needed after if

File: src/Scripts/azureHelper.ps1 [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    {
+        Write-Error "AZURE_CONFIG_DIR environment variable is empty, will not login"
+        exit -1
+    }
+
+    & az account show 2>&1 | Out-Null
+    if($LASTEXITCODE -ne 0)
```
Comment (richard-green) [inline src/Scripts/azureHelper.ps1:49]: whitespace needed after if

File: src/Scripts/azureHelper.ps1 [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+    & az webapp deployment source config-zip --resource-group $resourceGroup --name $name --src $zipPath
+}
+
+function Azure-DeployWebApp([string]$username, [string]$password, [string]$tenant, [string]$resourceGroup, [string]$name, [string]$contentFolder)
+{
+    Azure-EnsureProfileActive $username $tenant
+
+    Azure-LoginServicePrincipal $username $password $tenant
+
+    if(Test-Path "$contentFolder.zip")
```
Comment (richard-green) [inline src/Scripts/azureHelper.ps1:77]: whitespace needed after if

File: src/Scripts/webHelper.ps1 [hunk 11]
```diff
@@ -1,7 +1,12 @@
+if($deployHelpLoaded -eq $null)
```
Comment (richard-green) [inline src/Scripts/webHelper.ps1:1]: whitespace needed after if

**Review by richard-green — CHANGES_REQUESTED** [review]

File: src/Scripts/azureHelper.ps1 (hunk 2 above)
Comment (richard-green) [reply src/Scripts/azureHelper.ps1:73]: Again, the LoginServicePrincipal calls EnsureProfileActive for you

Comment (BlythMeister) [reply src/Scripts/azureHelper.ps1:73]: i'll remove, but the check was more there as a safety to ensure that beween login and use we are on the right profile.

Comment (BlythMeister) [reply src/Scripts/azureHelper.ps1:73]: oh i see what you mean, that check should of actually been on the other function.  but i'll just remove the call

**Review by richard-green — APPROVED** [review]

File: src/Scripts/azureHelper.ps1 [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+function Azure-DeployWebApp([string]$username, [string]$password, [string]$tenant, [string]$resourceGroup, [string]$name, [string]$contentFolder)
+{
+    if (Test-Path "$contentFolder.zip")
+    {
+        Remove-Item "$contentFolder.zip" -Force
+    }
+    
+    CreateZip $contentFolder "$contentFolder.zip"
+    
+    Azure-DeployWebApp $username $password $tenant $resourceGroup $name "$content.zip"
```
Comment (richard-green) [inline src/Scripts/azureHelper.ps1:82]: I think you mean $contentFolder.zip

**Review by richard-green — APPROVED** [review]

**Pull request merged**